provider/datagen/src/bin/datagen.rs

```diff
@@ -118,112 +113,112 @@ fn main() -> eyre::Result<()> {
                 .default_value("small"),
         )
         .arg(
-            Arg::with_name("COLLATION_HAN_DATABASE")
+            Arg::new("COLLATION_HAN_DATABASE")
                 .long("collation-han-database")
-                .takes_value(true)
-                .possible_values(&["unihan", "implicit"])
+                .num_args(1)
+                .value_parser(["unihan", "implicit"])
                 .default_value("implicit")
                 .help("Which collation han database to use.")
         )
         .arg(
-            Arg::with_name("INCLUDE_COLLATIONS")
+            Arg::new("INCLUDE_COLLATIONS")
                 .long("include-collations")
-                .multiple(true)
-                .takes_value(true)
-                .possible_values(&["gb2312", "big5han", "search", "searchjl", "search*"])
+                .action(clap::ArgAction::Append)
+                .num_args(1)
```

Comment:
Issue: Everything that was `.multiple(true).takes_value(true)` needs to accept syntax like `--locales en es zh` (infinite number of space-separated args)

Reply:
(resolved in latest commit)